docs/quickstart/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,890 @@
+- [Contract ABI Specification](https://docs.soliditylang.org/en/v0.8.10/abi-spec.html)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Customizing the EVM with Stateful Precompiles](https://medium.com/avalancheavax/customizing-the-evm-with-stateful-precompiles-f44a34f39efd)
+
+### The Process
+
+We will first create a Solidity interface that our precompile will implement.  Then we will use the precompile tool to autogenerate functions and fill out the rest. We're not done yet! We will then have to update a few more places within the Subnet-EVM. Some of this work involves assigning a precompile address, adding the precompile to the list of Subnet-EVM precompiles, and finally enabling the precompile. Now we can see our functions in action as we write another solidity smart contract that interacts with our precompile. Lastly, we will write some tests to make sure everything works as promised. We will also have an [official tutorial](https://github.com/ava-labs/subnet-evm/pull/290) with step by step commits you can follow to double check your work.
+
+## Tutorial
+
+We will first start off by creating the Solidity interface that we want our precompile to implement. This will be the HelloWorld Interface. It will have two simple functions, `sayHello` and `setGreeting`. These two functions will demonstrate the getting and setting respectively of a value using state access.
+
+We will place the interface in `./contract-examples/contracts`
+
+```sol
+// (c) 2022-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.0;
+
+interface IHelloWorld {
+ // sayHello returns the string located at [key]
+  function sayHello() external returns (string calldata result);
+
+// setGreeting sets the string located at [key]
+  function setGreeting(string calldata response) external;
+}
+```
+
+Now we have an interface that our precompile can implement!
+Let's create an [abi](https://docs.soliditylang.org/en/v0.8.13/abi-spec.html#:~:text=Contract%20ABI%20Specification-,Basic%20Design,as%20described%20in%20this%20specification.) of our solidity code.
+
+In the same `./contract-examples/contracts` directory, let's [download solc](https://docs.soliditylang.org/en/v0.8.9/installing-solidity.html) and run
+
+```bash
+solc --abi IHelloWorld.sol -o .
+```
+
+This spits out the abi code!
+
+IHelloWorld.abi
+
+``` json
+[{"inputs":[],"name":"sayHello","outputs":[{"internalType":"string","name":"result","type":"string"}],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"string","name":"recipient","type":"string"}],"name":"setGreeting","outputs":[],"stateMutability":"nonpayable","type":"function"}]
+```
+
+Note: The abi must have named outputs in order to generate the precompile template.
+
+## Generating the precompile
```

Comment:
```suggestion
### Generating the precompile
```
Should this section and all step sections be under Tutorial, rather parallel with it?

Reply:
Fixed.

Reply:
This does not look right to me.
<img width="344" alt="Screen Shot 2022-10-15 at 10 32 59 AM" src="https://user-images.githubusercontent.com/12171681/195997767-c24ad519-bb1b-40fd-b080-dc71a8befa54.png">

I was thinking

Tutorial 
  Generating the precompile
  Step 1
  ....
  Step 10

Reply:
Do you want the subcontents in background to be the same level?

Reply:
You should add **Tutorial** back as a separate section header and then put all sub-sections as subsection.

Reply:
Currently the **Tutorial** is not shown as a section header.

Reply:
Ok! thank you! Is this better? @yulin-dong

Reply:
Please verify 

<img width="366" alt="Screen Shot 2022-10-17 at 12 59 13 PM" src="https://user-images.githubusercontent.com/12171681/196260105-f6e69e02-e5cf-430c-9208-b8eae6e4f54a.png">

Reply:
Ah thanks